johnzon-jsonb/src/main/java/org/apache/johnzon/jsonb/JsonbAccessMode.java

```diff
@@ -1033,6 +1037,20 @@ private boolean isNumberType(final Type type) {
         return Number.class.isAssignableFrom(clazz) || clazz.isPrimitive();
     }
 
+    private boolean hasRawType(final Type type) {
+        return Class.class.isInstance(type) || ParameterizedType.class.isInstance(type);
+    }
+
+    private Class<?> getRawType(final Type type) {
+        if (!Class.class.isInstance(type) && !ParameterizedType.class.isInstance(type)) {
+            throw new IllegalStateException("Unsupported generic type " + type.getClass().getName());
+        } else if (Class.class.isInstance(type)) {
+            return Class.class.cast(type);
+        } else /*if (ParameterizedType.class.isInstance(type))*/ {
+            return getRawType(ParameterizedType.class.cast(type).getRawType());
```

Comment:
Cant it loop?

Reply:
Thinking about it, it neither needs recursion nor a loop since getRawType() always returns a class object

Reply:
Don't think so, Collection<Collection<Collection<....>>> ;)

Reply:
This would return Collection.class